clients/da-vinci-client/src/main/java/com/linkedin/davinci/repository/NativeMetadataRepository.java

```diff
@@ -75,14 +76,24 @@
   private final Set<StoreDataChangedListener> listeners = new CopyOnWriteArraySet<>();
   private final AtomicLong totalStoreReadQuota = new AtomicLong();
 
+  private final long refreshIntervalInSeconds;
+
+  private AtomicBoolean started = new AtomicBoolean(false);
```

Comment:
Can we add a defensive check in `subscribe(String storeName)` to fail explicitly if not started. This way the repository doesn't fail silently (serving stale metadata that will never get refreshed) if the dev forgot to call start() before using it.

Reply:
Good idea.

Reply:
done